packages/agent/src/core.ts

```diff
@@ -112,52 +107,61 @@ export default class AgentCore extends Store {
       raise('need generated id')
       return
     }
-    this.baseUrl = `${baseUrl}/${id}/${userId}`
+    this.baseUrl = `${settings.baseUrl}/${id}/${userId}`
     this.emitter.on(this.id, this.updateInteractCache.bind(this))
   }
 
-  send (type: HitType, sendData: SendData): void {
-    switch (type) {
-      case 'pageview':
-        this.sendInteracts(true)
-        if (!this.loadTime) {
-          this.bind()
-        }
+  pageview (page: string): void {
+    this.send([], true)
+    if (!this.loadTime) {
+      this.bind()
+    }
 
-        const data = getEnv(pathname2href(sendData as string))
-        if (!data || !this.baseUrl) {
-          return warning(`failed init`)
-        }
-        this.merge({ type: 'env', data })
-
-        this.interval = INTERVAL_DEFAULT_SETTING.concat()
-        this.interactId = 0
-        this.eventId = 0
-        this.loadTime = Date.now()
-        this.sendInteractsWithUpdate()
-        get(
-          `${this.baseUrl}/${this.loadTime}/env.gif`,
-          obj2query(objectAssign(
-            {},
-            this.get('env'),
-            this.get('custom')
-          ) as any),
-          () => {
-            this.destroy()
-          }
-        )
-        break
-      case 'event':
-        const event = sendData as SendEvent
-        if (
-          event.category &&
-          event.action &&
-          (!event.value || event.value >= 0)
-        ) {
-          this.eventId++
-          this.sendEvent(event)
-        }
-        break
+    const data = getEnv(pathname2href(page))
+    if (!data || !this.baseUrl) {
+      return warning(`failed init`)
+    }
+    this.merge({ type: 'env', data })
+
+    this.interval = INTERVAL_DEFAULT_SETTING.concat()
+    this.interactId = 0
+    this.eventId = 0
+    this.loadTime = Date.now()
+    this.sendWithUpdate()
+    get(
+      `${this.baseUrl}/${this.loadTime}/env.gif`,
+      obj2query(
+        objectAssign({}, this.get('env'), this.get('custom')) as any /* TODO */
+      ),
+      () => {
+        this.destroy()
+      }
+    )
+    this.set('page', undefined) // remove locale cache
```

Comment:
`state.env.l` is cached location

update without `send`, like `_ud('set', 'page', internet.url)` `_ud('set', { page: internet.url() })`